### PR TITLE
Added a user guide

### PR DESCRIPTION
**Pull request by etorreborre**
This is far from perfect but I think this is clearer than the previous doc because it is centered on the use of annotations from the start:

https://zalando.github.io/grafter

Review @jcranky @taojang @danielkarch

### CODE REVIEW

**Comment by danielkarch** [comment]
The picture does not load for me.

**Comment by etorreborre** [comment]
@danielkarch yes, just fixed that :-)

**Comment by danielkarch** [comment]
I like the Ascii-art application graphs.

File: guide/src/test/scala/org/zalando/grafter/QuickStart.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+### Installation
+
+To use Grafter you need add it as a dependency in your sbt build settings:
+
+```scala
+libraryDependencies += "org.zalando" %% "grafter" % "$version"
+
+// you also need the macros plugin for the grafter annotations
+addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full)
```
Comment (jcranky) [inline guide/src/test/scala/org/zalando/grafter/QuickStart.scala]: `2.1.1` or perhaps a more recent version?

**Comment by jcranky** [comment]
:+1:

File: guide/src/test/scala/org/zalando/grafter/QuickStart.scala (hunk 1 above)
Comment (etorreborre) [reply guide/src/test/scala/org/zalando/grafter/QuickStart.scala]: The macros site says `2.1.0`: http://docs.scala-lang.org/overviews/macros/paradise.html

Comment (jcranky) [reply guide/src/test/scala/org/zalando/grafter/QuickStart.scala]: Then they are outdated: http://search.maven.org/#artifactdetails%7Corg.scalamacros%7Cparadise_2.12.4%7C2.1.1%7Cjar

Comment (etorreborre) [reply guide/src/test/scala/org/zalando/grafter/QuickStart.scala]: Thanks I updated the version.

**Comment by jcranky** [comment]
:+1:

**Comment by etorreborre** [comment]
👍

**Pull request merged**